controllers/applicationvolumereplication_controller.go

```diff
@@ -120,106 +126,263 @@ func (r *ApplicationVolumeReplicationReconciler) Reconcile(ctx context.Context,
 				*subscription.Spec.Placement.Local) {
-			r.Log.Info("skipping local subscription", "name", subscription.Name)
+			r.Log.Info("Skipping local subscription", "name", subscription.Name)
 
 			continue
 		}
 
-		// Create a ManifestWork that represents the VolumeReplicationGroup CR for the hub to
-		// deploy on the managed cluster. A manifest workload is defined as a set of Kubernetes
-		// resources. The ManifestWork must be created in the cluster namespace on the hub, so that
-		// agent on the corresponding managed cluster can access this resource and deploy on the
-		// managed cluster. We create one ManifestWork for each VRG CR.
-		homeCluster, peerCluster, err := r.createVRGManifestWork(subscription)
-		if err != nil {
-			r.Log.Info("failed to create or update ManifestWork for", "subscription", subscription.Name, "error", err)
+		placementDecision, needRequeue := r.processSubscription(avr, &subscriptionList.Items[idx])
 
+		if needRequeue {
 			requeue = true
 
 			continue
 		}
 
-		placementDecisions[subscription.Name] = &ramendrv1alpha1.SubscriptionPlacementDecision{
-			HomeCluster: homeCluster,
-			PeerCluster: peerCluster,
+		if placementDecision != nil {
+			placementDecisions[subscription.Name] = placementDecision
 		}
-
-		r.Log.Info("created VolumeReplicationGroup manifest for ", "subscription", subscription.Name,
-			"HomeCluster", homeCluster, "PeerCluster", peerCluster)
 	}
 
+	r.Log.Info("Returning Placement Decisions", "Total", len(placementDecisions))
+
 	return placementDecisions, requeue
 }
 
-func (r *ApplicationVolumeReplicationReconciler) updateAVRStatus(
-	ctx context.Context,
+func (r *ApplicationVolumeReplicationReconciler) processSubscription(
 	avr *ramendrv1alpha1.ApplicationVolumeReplication,
-	placementDecisions ramendrv1alpha1.SubscriptionPlacementDecisionMap) error {
-	r.Log.Info("updating AVR status")
+	subscription *subv1.Subscription) (*ramendrv1alpha1.SubscriptionPlacementDecision, bool) {
+	const requeue = true
+	// Check to see if this subscription is paused for DR. If it is, then restore PVs to the new destination
+	// cluster, unpause the subscription, and skip it until the next reconciler iteration
+	if r.isSubsriptionPausedForDR(subscription.GetLabels()) {
+		if err := r.processPausedSubscription(avr, subscription); err != nil {
+			r.Log.Error(err, fmt.Sprintf("failed to process paused Subscription %s", subscription.Name))
+		}
+		// Stop processing this subscription. We'll wait for the next Reconciler iteration
+		r.Log.Info("Skipping paused subscription", "name", subscription.Name)
 
-	avr.Status = ramendrv1alpha1.ApplicationVolumeReplicationStatus{
-		Decisions: placementDecisions,
+		return nil, requeue
 	}
-	if err := r.Client.Status().Update(ctx, avr); err != nil {
-		return errorswrapper.Wrap(err, "failed to update AVR status")
+
+	// Skip this subscription if a manifestwork already exist
+	if found, err := r.findVRGManifestWork(avr, subscription); err != nil {
+		return nil, requeue
```

Comment:
(nit) Log error either here or in findVRGManifestWork?

Reply:
Yes, I was going to bring up this in the standup.  We need to standardize on logging location -- Log in the called method or Log at the caller if the call fails.  I vote for the earlier so that if a method is reused, you'll get a standard log.